src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -597,6 +599,38 @@ public void testAddRemoteUrl() throws Exception {
         assertEquals(upstreamRepoURL, gitClient.getRemoteUrl("upstream"));
     }
 
+    @Test
+    public void testAutocreateFailsOnMultipleMatchingOrigins() throws Exception {
+        File repoRootTemp = tempFolder.newFolder();
+        GitClient gitClientTemp = Git.with(TaskListener.NULL, new EnvVars()).in(repoRootTemp).using(gitImplName).getClient();
+        gitClientTemp.init();
+        gitClientTemp.clone_().url("file://" + repoRootTemp.getPath()).execute();
+        final URIish remote = new URIish(Constants.DEFAULT_REMOTE_NAME);
+
+        try ( // add second remote
+              FileRepository repo = new FileRepository(repoRootTemp)) {
+            StoredConfig config = repo.getConfig();
+            config.setString("remote", "upstream", "url", "file://" + repoRootTemp.getPath());
+            config.setString("remote", "upstream", "fetch", "+refs/heads/*:refs/remotes/upstream/*");
+            config.save();
+        }
+
+        // fill both remote branches
+        List<RefSpec> refspecs = Collections.singletonList(new RefSpec(
+                "refs/heads/*:refs/remotes/origin/*"));
+        gitClientTemp.fetch_().from(remote, refspecs).execute();
+        refspecs = Collections.singletonList(new RefSpec(
+                "refs/heads/*:refs/remotes/upstream/*"));
+        gitClientTemp.fetch_().from(remote, refspecs).execute();
+
+        try {
+            gitClientTemp.checkout().ref(Constants.MASTER).execute();
+            fail("GitException expected");
+        } catch (GitException e) {
+            // expected
```

Comment:
I fear that the exception in this case is probably different than the exception in the test that has been deleted.  Unfortunately, the test is not asserting enough about the exception.  Would you be willing to improve the original test by asserting the contents of the exception?

Reply:
Yes, I am happy to improve by writing some assertion about the exception.